app/main/views.py

```diff
@@ -104,21 +104,28 @@ def update(service_id, section):
         list(request.form.items()) + list(request.files.items())
     )
 
-    errors = Validate(content, service, posted_data, s3_uploader).errors
+    form = Validate(content, service, posted_data, s3_uploader)
+    update = {}
 
     for question_id in posted_data:
-        if question_id not in errors:
-            service_loader.set(service, question_id, "new value")
-
-    service_loader.post(service)
+        if question_id not in form.errors and question_id in form.clean_data:
+            update[question_id] = form.clean_data[question_id]
+
+    if update:
+        service_loader.post(
```

Comment:
If we always timestamp new, rather than previous files (as discussed yesterday, sorry @robyoung ) then inconsistency is unpossible.